src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandCanvasVisitor.java

```diff
@@ -20,15 +20,13 @@ public CommandCanvasVisitor(Canvas canvas) {
     @Override
     public void visitOperateToCommand(OperateToCommand operateToCommand) {
         Point pointToCheck = operateToCommand.getPoint();
-        if(!canvas.checkIfPointIsOnCanvas(pointToCheck))
-            allPointsOnCanvas = false;
+        allPointsOnCanvas = !canvas.checkIfPointIsOnCanvas(pointToCheck);
     }
 
     @Override
     public void visitSetPositionCommand(SetPositionCommand setPositionCommand) {
         Point pointToCheck = setPositionCommand.getPoint();
-        if(!canvas.checkIfPointIsOnCanvas(pointToCheck))
-            allPointsOnCanvas = false;
+        allPointsOnCanvas = !canvas.checkIfPointIsOnCanvas(pointToCheck);
     }
```

Comment:
Without modifications to visitICompoundCommand I don't think it would work correctly. There are no tests that confirm the correctness of the solution.